src/components/v1/V1Project/ProjectActivity/index.tsx

```diff
@@ -7,321 +7,120 @@ import DeployedERC20EventElem from 'components/activityEventElems/DeployedERC20E
-            elem = (
-              <ReservesEventElem
-                event={e.printReservesEvent as PrintReservesEvent}
-              />
-            )
-          }
-          if (e.projectCreateEvent) {
-            elem = (
-              <ProjectCreateEventElem
-                event={e.projectCreateEvent as ProjectCreateEvent}
-              />
-            )
-          }
-          if (e.deployedERC20Event) {
-            elem = (
-              <DeployedERC20EventElem
-                event={e.deployedERC20Event as DeployedERC20Event}
-              />
-            )
-          }
-          if (e.v1ConfigureEvent) {
-            elem = (
-              <V1ConfigureEventElem
-                event={e.v1ConfigureEvent as V1ConfigureEvent}
-              />
-            )
-          }
-
-          if (!elem) return null
-
-          return (
-            <div
-              className="mb-5 border-b border-smoke-200 pb-5 dark:border-grey-600"
-              key={e.id}
-            >
-              {elem}
-            </div>
-          )
-        }),
-      ),
+      projectEvents?.map(e => {
+        let elem: JSX.Element | undefined = undefined
+
+        if (e.payEvent) {
+          elem = <PayEventElem event={e.payEvent} />
+        }
+        if (e.burnEvent) {
+          elem = <BurnEventElem event={e.burnEvent} tokenSymbol={tokenSymbol} />
+        }
+        if (e.addToBalanceEvent) {
+          elem = <AddToBalanceEventElem event={e.addToBalanceEvent} />
+        }
+        if (e.tapEvent) {
+          elem = <TapEventElem event={e.tapEvent} />
+        }
+        if (e.redeemEvent) {
+          elem = <RedeemEventElem event={e.redeemEvent} />
+        }
+        if (e.printReservesEvent) {
+          elem = <ReservesEventElem event={e.printReservesEvent} />
+        }
+        if (e.projectCreateEvent) {
+          elem = <ProjectCreateEventElem event={e.projectCreateEvent} />
+        }
+        if (e.deployedERC20Event) {
+          elem = <DeployedERC20EventElem event={e.deployedERC20Event} />
+        }
+        if (e.v1ConfigureEvent) {
+          elem = <V1ConfigureEventElem event={e.v1ConfigureEvent} />
+        }
+
+        if (!elem) return null
+
+        return (
+          <div
+            className="mb-5 border-b border-smoke-200 pb-5 dark:border-grey-600"
+            key={e.id}
+          >
+            {elem}
+          </div>
+        )
+      }),
     [projectEvents, tokenSymbol],
   )
 
   const listStatus = useMemo(() => {
-    if (isLoading || isFetchingNextPage) {
+    if (loading) {
       return (
         <div>
           <Loading />
         </div>
       )
     }
 
-    if (count === 0 && !isLoading) {
+    if (count === 0 && !loading) {
       return (
         <div className="border-t border-smoke-200 pt-5 text-grey-500 dark:border-grey-600 dark:text-grey-300">
           <Trans>No activity yet</Trans>
         </div>
       )
     }
 
-    if (hasNextPage) {
+    if (count % PAGE_SIZE === 0) {
```

Comment:
awkward. works, except for when the activity event count is divisible by PAGE_SIZE :/